src/CcdImage.cc

```diff
@@ -55,27 +55,27 @@ void CcdImage::LoadCatalog(const lsst::afw::table::SortedCatalogT<lsst::afw::tab
         ms->flux = i->get(fluxKey);
         ms->eflux = i->get(efluxKey);
         ms->mag = -2.5*log10(ms->flux) + zp;
-        ms->SetCcdImage(this);
+        ms->setCcdImage(this);
         wholeCatalog.push_back(ms);
     }
-    wholeCatalog.SetCcdImage(this);
+    wholeCatalog.setCcdImage(this);
 }
 
 
 
-static int getBandIndex(std::string const& band)
+static int findFilterIndex(std::string const& filter)
 {
-    if (band == "u") return 0;
-    if (band == "g") return 1;
-    if (band == "r") return 2;
-    if (band == "i") return 3;
-    if (band == "z") return 4;
-
-    if (band == "U") return 5;
-    if (band == "B") return 6;
-    if (band == "V") return 7;
-    if (band == "R") return 8;
-    if (band == "I") return 9;
+    if (filter == "u") return 0;
+    if (filter == "g") return 1;
+    if (filter == "r") return 2;
+    if (filter == "i") return 3;
+    if (filter == "z") return 4;
+
+    if (filter == "U") return 5;
+    if (filter == "B") return 6;
+    if (filter == "V") return 7;
+    if (filter == "R") return 8;
+    if (filter == "I") return 9;
```

Comment:
Please consider a case statement. It would be more idiomatic and might be more efficient. A `map` might be even more efficient (I don't know) but would be more work.